### PR TITLE
Created documentation for `MiddleTextTruncation`

### PR DESCRIPTION
**Pull request by smmr-dn**
## Changes

Created documentation for `MiddleTextTruncation`.

## Testing

N/A

## Docs

N/A

### CODE REVIEW

File: apps/website/src/content/docs/middletexttruncation.mdx [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+
+The function should return a `ReactNode` that represents the customized text element. This is particularly useful when end users want to apply specific styles or add additional elements to the truncated text, such as highlighting the truncated portion or adding tooltips for better user experience.
+
+<LiveExample src='MiddleTextTruncation.customizeTruncation.jsx'>
+  <AllExamples.MiddleTextTruncationCustomizationExample client:load />
+</LiveExample>
+
+### Accessibility
+
+This component is designed for visual truncation and should not impact the accessibility tree, as the truncated text would be confusing, especially for non-sighted users. The original text is kept visually hidden, and the truncated text is removed from the accessibility tree. This approach ensures consistency with the built-in `text-overflow: ellipsis`, allowing the same UX design guidelines to be applied in both cases.
```
[the remark this answers is not in the document]
Comment (smmr-dn) [reply apps/website/src/content/docs/middletexttruncation.mdx]: All the wording suggestions above make sense to me, so I have made all changes.

File: apps/website/src/content/docs/middletexttruncation.mdx [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+### Customization
+
+For further control over the rendering of the content, the `textRenderer` prop allows users to provide a custom rendering function. This function takes `originalText` and `truncatedText` as parameters and returns a `ReactNode`.
+
+<LiveExample src='MiddleTextTruncation.customizeTruncation.jsx'>
+  <AllExamples.MiddleTextTruncationCustomizationExample client:load />
+</LiveExample>
+
+### Accessibility
```
Comment (mayank99) [inline apps/website/src/content/docs/middletexttruncation.mdx:36]: This doesn't belong under "Usage" so it should be an h2

```suggestion
## Accessibility
```

File: apps/website/src/content/docs/middletexttruncation.mdx [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+
+This component only truncates the content visually but leaves the full content in the accessibility tree. That way, users using assistive technologies can still have access to the full content which is consistent with the default overflow behavior.
```
Comment (mayank99) [inline apps/website/src/content/docs/middletexttruncation.mdx:38]:
```suggestion
This component only truncates the content visually but leaves the full content in the accessibility tree. That way, users using assistive technologies can still have access to the full content. This is also consistent with the browser's default `text-overflow` behavior.
```

**Review by mayank99 — APPROVED** [review]

File: apps/website/src/content/docs/middletexttruncation.mdx (hunk 3 above)
Comment (smmr-dn) [reply apps/website/src/content/docs/middletexttruncation.mdx]: I have made the change.

File: apps/website/src/content/docs/middletexttruncation.mdx [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+
+## Usage
+
+In circumstances where a fixed width constraint is required, the `MiddleTextTruncation` component can be helpful in handling long text strings by truncating them in the middle by inserting the ellipsis (`"..."`). This ensures that the beginning and the end of the text remain visible.
+
+<LiveExample src='MiddleTextTruncation.main.jsx' truncated={false}>
+  <AllExamples.MiddleTextTruncationMainExample client:load />
+</LiveExample>
+
+For more application examples, see this [documentation on Select](/docs/select#truncation).
```
[the remark this answers is not in the document]
Comment (smmr-dn) [reply apps/website/src/content/docs/middletexttruncation.mdx]: I have made the change.

**Pull request merged**